src/ui/DynamicFacet/DynamicFacet.ts

```diff
@@ -410,6 +402,31 @@ export class DynamicFacet extends Component {
     this.dynamicFacetQueryController.enableFreezeFacetOrderFlag();
   }
 
+  public get analyticsFacetState(): IAnalyticsDynamicFacetMeta[] {
+    return this.values.activeFacetValues.map(facetValue => facetValue.analyticsMeta);
+  }
+
+  public logAnalyticsEvent(actionCause: IAnalyticsActionCause, facetMeta: IAnalyticsDynamicFacetMeta) {
+    this.usageAnalytics.logSearchEvent<IAnalyticsDynamicFacetMeta>(actionCause, facetMeta);
+  }
+
+  /**
+   * Adds the state of the facet inside the query.
+   * @param queryBuilder [`QueryBuilder`]{@link QueryBuilder}
+   */
+  public putStateIntoQueryBuilder(queryBuilder: QueryBuilder) {
```

Comment:
My comments for public documentation was valid for all of those method.

A developer using this component would probably never need to call any of those methods directly, or to even know they exists.

Rule of thumb: A component should expose and document public methods that loosely mimic/reflect what action an actual end user would like to perform against the component. This makes for a "clean" API to use through code.

A component where external code has to manually "manage the query" or analytics is not easy nor simple to use.

Reply:
I hesitated removing this one because its a bit old (part of the last release) but I realise new doc hasnt been generated yet so I guess I can still remove it